### PR TITLE
fix(curriculum): cat photo app step 31 to 35

### LINKED ISSUE #56315: Update steps 31-35 of cat photo workshop to use specific assert methods where appropriate
We recently updated the guidelines for writing assertions here 
https://contribute.freecodecamp.org/how-to-work-on-coding-challenges/#writing-tests

For consistency and matching what is in the docs, we should slowly start to update the tests for the workshops and labs in the new frontend cert.

For this issue, just focus on updating the tests for steps 31-35 of the cat photo app workshop.
We don't want to have a large PR that updates a whole bunch of tests in a bunch of steps and makes it harder to review. 

Here is the link to the workshop:
https://github.com/freeCodeCamp/freeCodeCamp/tree/main/curriculum/challenges/english/25-front-end-development/workshop-cat-photo-app

The tests should be updated, where appropriate, to use specific methods like `assert.equal`, `assert.exists`, `assert.match`, etc instead of just `assert`.

Here is the complete API reference 
https://www.chaijs.com/api/assert/#method_exists

Just focus on the new frontend cert and don't worry about updating the RWD cat photo app since that will become legacy anyway.

### PR DESCRIPTION
**Pull request by gagan-bhullar-tech**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/how-to-open-a-pull-request/).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #56315

### CODE REVIEW

File: curriculum/challenges/english/25-front-end-development/workshop-cat-photo-app/6698491eff5f2e43c20c6503.md [hunk 1]
```diff
@@ -14,66 +14,65 @@ After the last `img` element, add a `figcaption` element with the text `Cats hat
 Your `figcaption` element should have an opening tag. Opening tags have the following syntax: `<elementName>`.
 
 ```js
-assert(document.querySelectorAll('figcaption').length === 2);
+assert.lengthOf(document.querySelectorAll('figcaption').length, 2);
```
Comment (ilenia-magoni) [inline curriculum/challenges/english/25-front-end-development/workshop-cat-photo-app/6698491eff5f2e43c20c6503.md:17]: same here, check all cases of `lengthOf`

File: curriculum/challenges/english/25-front-end-development/workshop-cat-photo-app/66984ec64a3652522e81c678.md [hunk 2]
```diff
@@ -14,29 +14,30 @@ Turn the existing `freeCodeCamp.org` text into a link by enclosing it in an anch
 Your anchor (`a`) element should be nested within the `p` element inside the `footer`. Make sure to add an opening tag and closing tag for the anchor (`a`) element.
 
 ```js
-assert($('footer > p > a').length);
+assert.isNotEmpty($('footer > p > a').length);
```
Comment (ilenia-magoni) [inline curriculum/challenges/english/25-front-end-development/workshop-cat-photo-app/66984ec64a3652522e81c678.md:17]:
```suggestion
assert.isNotEmpty(document.querySelectorAll('footer > p > a'));
```

**Review by a2937 — APPROVED** [review]
All the tests pass and the assertions seem okay as far as I can tell.

File: curriculum/challenges/english/25-front-end-development/workshop-cat-photo-app/6698491eff5f2e43c20c6503.md [hunk 3]
```diff
@@ -14,66 +14,65 @@ After the last `img` element, add a `figcaption` element with the text `Cats hat
...
 ```js
-assert(code.match(/<\/figcaption\>/g).length === 2);
+assert.lengthOf(code.match(/<\/figcaption\>/g), 2);
 ```
 
 There should be a `figure` element right above the second `section` element's closing tag.
 
 ```js
-assert($('main > section')[1].lastElementChild.nodeName === 'FIGURE');
+assert.equal($('main > section')[1].lastElementChild.nodeName, 'FIGURE');
```
Comment (a2937) [inline curriculum/challenges/english/25-front-end-development/workshop-cat-photo-app/6698491eff5f2e43c20c6503.md:29]: Oh. I should have seen that.

File: curriculum/challenges/english/25-front-end-development/workshop-cat-photo-app/66984e8ab54bee517207be42.md [hunk 4]
```diff
@@ -14,27 +14,29 @@ Nest a `p` element with the text `No Copyright - freeCodeCamp.org` within the `f
...
 const pElemClosingTags = code.match(/<\/p\>/g);
-assert(pElemClosingTags && pElemClosingTags.length === 2);
+assert.exists(pElemClosingTags);
+assert.lengthOf(pElemClosingTags, 2);
 ```
 
 Your `p` element's text should be `No Copyright - freeCodeCamp.org`. You have either omitted the text, have a typo, or it is not between the `p` element's opening and closing tags.
 
 ```js
 const extraSpacesRemoved = $('footer > p')[0].innerText.replace(/\s+/g, ' ');
```
Comment (a2937) [inline curriculum/challenges/english/25-front-end-development/workshop-cat-photo-app/66984e8ab54bee517207be42.md:38]: There's a jQuery call here

File: curriculum/challenges/english/25-front-end-development/workshop-cat-photo-app/66984ec64a3652522e81c678.md [hunk 5]
```diff
@@ -14,29 +14,30 @@ Turn the existing `freeCodeCamp.org` text into a link by enclosing it in an anch
...
 const aElemClosingTags = code.match(/<\/a\>/g);
-assert(aElemClosingTags && aElemClosingTags.length === 3);
+assert.exists(aElemClosingTags);
+assert.lengthOf(aElemClosingTags, 3);
 ```
 
 Your anchor (`a`) element should have an `href` attribute with the value `https://www.freecodecamp.org`. You may have omitted the attribute/value, or have a typo.
 
 ```js
 const nestedAnchor = $('footer > p > a')[0];
```
Comment (a2937) [inline curriculum/challenges/english/25-front-end-development/workshop-cat-photo-app/66984ec64a3652522e81c678.md:31]: Here as well

File: curriculum/challenges/english/25-front-end-development/workshop-cat-photo-app/66984ec64a3652522e81c678.md [hunk 6]
```diff
@@ -14,29 +14,30 @@ Turn the existing `freeCodeCamp.org` text into a link by enclosing it in an anch
...
-assert(nestedAnchor.getAttribute('href').toLowerCase() === 'https://www.freecodecamp.org');
+assert.equal(nestedAnchor.getAttribute('href').toLowerCase(), 'https://www.freecodecamp.org');
 ```
 
 The link's text should be `freeCodeCamp.org`. You have either omitted the text or have a typo.
 
 ```js
 const nestedAnchor = $('footer > p > a')[0];
```
Comment (a2937) [inline curriculum/challenges/english/25-front-end-development/workshop-cat-photo-app/66984ec64a3652522e81c678.md:38]: Another jQuery call

File: curriculum/challenges/english/25-front-end-development/workshop-cat-photo-app/66984ec64a3652522e81c678.md [hunk 7]
```diff
@@ -45,7 +46,7 @@ After nesting the anchor (`a`) element, the only `p` element content visible in
 
 ```js
 const pText = $('footer > p')[0].innerText.toLowerCase().replace(/\s+/g, ' ');
```
Comment (a2937) [inline curriculum/challenges/english/25-front-end-development/workshop-cat-photo-app/66984ec64a3652522e81c678.md:48]: Here's another jQuery call.

**Review by a2937 — CHANGES_REQUESTED** [review]
I apologize for the early approval. Remove the jquery calls  please.

**Comment by gagan-bhullar-tech** [comment]
@a2937 updated PR.

**Review by majestic-owl448 — APPROVED** [review]
LGTM 👍

**Review by a2937 — APPROVED** [review]
Looks good to me.

**Pull request merged**